rlclientlib/live_model_impl.cc

```diff
@@ -598,20 +598,13 @@ namespace reinforcement_learning {
   }
 
   int reset_action_order(ranking_response& response) {
-#ifdef __clang__
     std::vector<action_prob> tmp;
```

Comment:
We try to avoid allocations when possible.  So this is undesirable.  I believe this was introduced to fix a Mac issue but the desirable approach is to not create a vector here.   It's better to do the sort on the existing collection.

Reply:
That's not possible because the `response` collection is not sortable. The standard requires a random access iterator and `response::begin`  returns a `container_iterator` which is tagged as a `std::forward_iterator_tag`.

Some compilers enforce the tag while others just take anything that implements all required methods. C++ is beautiful.

An alternative solution would be to change the tag in `container_iterator` to random access, but that would be a breaking change.

Reply:
Would that breaking change work for you, Jakub?